cpp/oneapi/dal/backend/primitives/reduction/functors.hpp

```diff
@@ -99,6 +131,22 @@ struct min {
     }
 };
 
+template <typename T>
+struct logical_or {
+    using tag_t = reduce_binary_op_tag;
+    constexpr static inline T init_value = false;
+#ifdef __SYCL_DEVICE_ONLY__
+    constexpr static inline sycl::logical_or<T> native{};
+#else
+    constexpr static inline auto native = [](const T& a, const T& b) {
+        return std::logical_or(a, b);
+    };
```

Comment:
Why cannot it be done the same way as in SYCL? It seems std::logical_or is also a template.